File: src/main/java/com/sup/dev/java_pc/views/widgets/ZMenuItem.java
package com.sup.dev.java_pc.views.widgets;

import com.sup.dev.java.classes.callbacks.simple.Callback1;
import com.sup.dev.java_pc.views.GUI;

import javax.swing.*;
import java.awt.*;

public class ZMenuItem extends JMenuItem {


    public ZMenuItem(String text, Callback1<String> callback){
        setBackground(Color.WHITE);
        setFont(GUI.BODY_2);
        setText(text);

        addActionListener(e ->  callback.callback(text));

    }

}

File: src/main/java/com/sup/dev/java_pc/views/widgets/ZMenuItem.kt
package com.sup.dev.java_pc.views.widgets

import com.sup.dev.java_pc.views.GUI
import java.awt.Color
import javax.swing.JMenuItem


class ZMenuItem(text: String?, callback: (String?)->Unit) : JMenuItem() {


    init {
        background = Color.WHITE
        font = GUI.BODY_2
        setText(text)

        addActionListener { e -> callback.invoke(text) }

    }

}
